x/datadeal/types/datasale.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func NewDataSale(msg *MsgSellData) *DataSale {
+	return &DataSale{
+		SellerAddress: msg.SellerAddress,
+		DealId:        msg.DealId,
+		VerifiableCid: msg.VerifiableCid,
+		Status:        DATA_SALE_STATUS_VERIFICATION_VOTING_PERIOD,
+	}
+}
+
+func (m DataSale) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.SellerAddress); err != nil {
+		return sdkerrors.Wrapf(err, "seller address is invalid. address: %s", m.SellerAddress)
+	}
+
+	if m.DealId == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "dealID should be bigger than 0")
+	}
+
+	if m.VerifiableCid == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "verifiable CID is empty")
+	}
+
+	if m.DeliveredCid == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "delivered CID is empty")
+	}
+
+	if m.VotingPeriod == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "votingPeriod is nil")
+	}
```

Comment:
This code same too.

Reply:
I fixed and applied your review in [a980c98](https://github.com/medibloc/panacea-core/pull/421/commits/a980c9882d53f02b037e4e13138479378d90eac3).